src/utils/set.ts

```diff
@@ -24,17 +28,38 @@ module Plottable {
         var index = this._values.indexOf(value);
         if (index !== -1) {
           this._values.splice(index, 1);
+          this._setSize(this._values.length);
           return true;
         }
         return false;
       }
 
+      private _setSize(size: number) {
+        Object.defineProperty(this, "size", {
+          value: this._values.length,
+          configurable: true,
+          writable: false
+        });
+        if (this.size !== size) {
```

Comment:
Can we more direct and put a line like
`if (isPhantomJS()) { this.size = size }`

We can even use can if-else to make it more clear that we only use this idea if we are in phantomJS